### PR TITLE
docs: add istio docs

### LINKED ISSUE #11821: Documentation request: please add this solution for coder's embedded DERP server behind an istio ingress reverse proxy
I am very happy with the 'health' tab in the coder ui, which revealed to me that the embedded derp server was not accessible. The help documentation which I was then pointed to, was helpful as well. If you could add the following solution somehow somewhere, this will save all people with the same problem the two days it took me to find this solution.

When coder is installed inside an istio mesh, and your ingress is managed by istio, you have to apply the following manifest:
```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: headscale-behind-istio-ingress
  namespace: istio-system
spec:
  configPatches:
    - applyTo: NETWORK_FILTER
      match:
        listener:
          filterChain:
            filter:
              name: envoy.filters.network.http_connection_manager
      patch:
        operation: MERGE
        value:
          typed_config:
            "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
            upgrade_configs:
              - upgrade_type: derp
```

which is directly derived from the solution found at https://github.com/juanfont/headscale/blob/main/docs/reverse-proxy.md#istio

### PR DESCRIPTION
**Pull request by ericpaulsen**
closes https://github.com/coder/coder/issues/11821

### CODE REVIEW

File: docs/manifest.json [hunk 1]
```diff
@@ -696,6 +696,11 @@
 					"description": "Integrate Coder with JFrog Artifactory",
 					"path": "./admin/integrations/jfrog-artifactory.md"
 				},
+				{
+					"title": "Istio Integration",
```
Comment (EdwardAngert) [inline docs/manifest.json:700]:
```suggestion
					"title": "Integrate Coder with Istio",
```

to match the ones around it

File: docs/tutorials/istio.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Configure Istio Service Mesh
```
Comment (EdwardAngert) [inline docs/tutorials/istio.md:1]:
```suggestion
# Integrate Coder with Istio
```

Next will use whatever manifest.json tells it to, so the H1 here feels like it's more of just a reference for contributors

File: docs/tutorials/istio.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Integrating Istio's service mesh with Coder's Ingress enables powerful traffic management, security, and observability capabilities. By placing Coder's workspace traffic behind Istio's intelligent proxy layer, you can implement access controls, encrypt service-to-service communication, and gain visibility into your workspace network patterns. This guide walks through the process of configuring Istio alongside Coder's existing ingress controller, ensuring that developer workspaces remain accessible while benefiting from Istio's comprehensive service mesh features.
+
+Before proceeding, ensure you have a running Kubernetes cluster with both Coder and Istio installed, and that you have administrative access to configure both systems. Once you have access to your Coder cluster, apply the following manifest:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
```
Comment (EdwardAngert) [inline docs/tutorials/istio.md:8]: I wasn't sure about putting an `alpha` version in our docs so I had to hunt a bit to confirm the version - seems old but Istio's docs still refer to it <https://istio.io/latest/docs/reference/config/networking/envoy-filter/>

File: docs/tutorials/istio.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Configure Istio Service Mesh
+
+Integrating Istio's service mesh with Coder's Ingress enables powerful traffic management, security, and observability capabilities. By placing Coder's workspace traffic behind Istio's intelligent proxy layer, you can implement access controls, encrypt service-to-service communication, and gain visibility into your workspace network patterns. This guide walks through the process of configuring Istio alongside Coder's existing ingress controller, ensuring that developer workspaces remain accessible while benefiting from Istio's comprehensive service mesh features.
```
Comment (EdwardAngert) [inline docs/tutorials/istio.md:3]:
```suggestion
Use Istio service mesh for your Coder workspace traffic to implement access controls, encrypt service-to-service communication, and gain visibility into your workspace network patterns.

This configures Istio alongside Coder's existing ingress controller, ensuring that developer workspaces remain accessible while benefiting from Istio's comprehensive service mesh features.
```

File: docs/tutorials/istio.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+Before proceeding, ensure you have a running Kubernetes cluster with both Coder and Istio installed, and that you have administrative access to configure both systems. Once you have access to your Coder cluster, apply the following manifest:
```
Comment (EdwardAngert) [inline docs/tutorials/istio.md:5]:
```suggestion
Ensure you have a running Kubernetes cluster with both Coder and Istio installed, and that you have administrative access to configure both systems.

Once you have access to your Coder cluster, apply the following manifest:
```

**Review by EdwardAngert — CHANGES_REQUESTED** [review]

File: docs/tutorials/istio.md (hunk 4 above)
Comment (johnstcn) [reply docs/tutorials/istio.md:3]: Coder does not have an existing ingress controller in the Kubernetes sense of the word.

Also, we could probably just drop the first sentence and get straight to the point:

```
This guide walks through the required steps to configure the Istio service mesh for use with Coder.
```

File: docs/tutorials/istio.md (hunk 5 above)
Comment (johnstcn) [reply docs/tutorials/istio.md:5]: Istio is theoretically platform-independent but in practice most folks use it with Kubernetes. We can just state this assumption outright.

File: docs/tutorials/istio.md (hunk 3 above)
Comment (johnstcn) [reply docs/tutorials/istio.md:8]: Yep, this is common in Kubernetes-land. I'm not familiar enough with Istio / Envoy to know if there is a different way to accomplish the same task using `networking.istio.io/v1` APIs.

**Review by EdwardAngert — APPROVED** [review]

**Review by johnstcn — APPROVED** [review]

**Pull request merged**